nestkernel/node.h

```diff
@@ -238,33 +238,29 @@ class Node
   void set_node_uses_wfr( const bool );
 
   /**
-   * Set state variables to the default values for the model.
+   * Set state variables to the default values for the model and initialize
+   * buffers of the node.
+   *
    * Dynamic variables are all observable state variables of a node
-   * that change during Node::update().
-   * After calling init_state(), the state variables
-   * should have the same values that they had after the node was
-   * created. In practice, they will be initialized to the values
-   * of the prototype node (model).
-   * @note If the parameters of the model have been changed since the node
-   *       was created, the node will be initialized to the present values
-   *       set in the model.
-   * @note This function is the public interface to the private function
-   *       Node::init_state_(const Node&) that must be implemented by
-   *       derived classes.
-   */
-  void init_state();
-
-  /**
-   * Initialize buffers of a node.
-   * This function initializes the Buffers of a Node, e.g., ring buffers
-   * for incoming events, buffers for logging potentials.
-   * This function is called before Simulate is called for the first time
+   * that change during Node::update(). After calling init(), the state
+   * variables should have the same values that they had after the node
+   * was created. In practice, they will be initialized to the values
```

Comment:
Calls to init() after the first call will have no effect due to ``if ( state_buffers_initialized_ ) return;`` — I cannot find a spot where we do ``state_buffers_initialized_ = False``, for instance at the end of set_status().

Reply:
The comment here is misleading/outdated. It described intended behaviour at a time when we had `ResetNetwork`. Now, `init()` only ensures that state variables which require dynamic configuration and buffers are configured properly. `SetStatus` can later change the values of state variables, but does not require `init()`.

Reply:
Here a suggestion for a new version of this comment, taking into account that we dropped `ResetNetwork`.

----------

Initialize node prior to first simulation after node has been created.

init() allows the node to configure internal data structures prior to being simulated. The method has an effect only the first time it is called on a given node, otherwise it returns immediately. init() calls virtual functions init_state_() and init_buffers_(). 

--------

Here `init_state_()` and `init_buffers_()` should ideally be links to the pertaining functions. 

Suggestions for their respective docstrings:

init_state

---------

Configure state variables depending on runtime information.

Overload this method if the node needs to adapt state variables prior to first simulation to runtime information, e.g., the number of incoming connections.

------------

init_buffers

----------

Configure persistent internal data structures.

Let node configure persistent internal data structures, such as input buffers or ODE solvers, to runtime information prior to first simulation.

--------------

Reply:
I have updated the comments.